week-10/10-tech-lab.adoc

```diff
@@ -0,0 +1,228 @@
+
+=== System hardening
+
+Now for the hard part. Find another vulnerability you can fix. To use class time most efficiently, you should post the vulnerabilities you are researching to Slack. Don't start on something someone else is doing.
+
+You should take some time and think about and/or research the risk any given vulnerability presents. Remember the CISSP terminology:
+
+image::4.10-security-taxonomy.png[]
+
+We will pause class and discuss interesting cases.
+
+TIP: If you search Google on the Lynis ID, e.g. `AUTH-9286`, you should be taken to the cisofy.com site (or perhaps other resources) which will get you started on resolving it.
+
+NOTE: Do not try to fix the banners issue. It only can be tested by logging directly into the server console (which in the case of a VM doesn't really exist), not via ssh.
+
+Be prepared to share you results, positive or negative. If you fix a vulnerability, be sure to document the steps so they could be scripted.
+
+After you have fixed one more lynis vulnerability, read http://www.sonatype.org/nexus/2015/11/13/did-you-wake-up-to-an-alert-about-the-java-deserialization-vulnerability and discuss with your neighbors. What is the National Vulnerability Database? Be prepared to discuss in class.
+
+
+== Permanently fixing one or more vulnerabilities (10+ extra credit points).
+
+First, you need to figure out the configuration change required to fix the vulnerability. It should be something that you can script.
+
+* something installed that doesn't need to be
+* something that should be there, that isn't
+* permissions that need to be tightened down.
+
+* 1 vulnerability fixed: 10 points
+* 2 : 15 points
+* 3 : 20 points.
+* 10 or more: 30 points
+
+In order to gain the extra credit:
+
+* Create a new VM and run the vulnerability scanner.
+* Save the output to a text file, named "initial-output.log"
+* Run the necessary commands to fix the vulnerability.
+* Put them into a shell script.
+* Run the scanner again, saving the output to "final-output.log" and showing that they are fixed
+
+IMPORTANT: Submit the 2 output files, and your script for extra credit.
+
+You can script commands like
+
+ apt-get install <some program>
+
+You can't script an interactive text editor (like nano). To change configuration files, sed is your friend. Simple example:
+
+....
+char@seis664:~$ echo myScriptContent > test.txt
+char@seis664:~$ cat test.txt
+myScriptContent
+char@seis664:~$ sed -i s/Script/New/g test.txt
+char@seis664:~$ cat test.txt
+myNewContent
+....
+
+You may need to Google "regular expressions."
+
+=== Web vulnerability testing scan
+(if time)
+
+When you have reflected enough on the Java deserialization vulnerability and the implications of that article, move on to a Web vulnerability. We will install 
+Java and apache2 on your VM. (Don't try to install these on the main server; you won't be able to.)
```

Comment:
We don't use a "main server" anymore. That's the kind of thing I need cleaned up.

Reply:
they can use the same VM, mention this.

Reply:
Fixed.